[{"name": "test.test_rational.RationalTest.test_comparison", "status": "errored", "message": "name 'common_denominator' is not defined", "passed": false, "points": ["p02-09.1"], "backtrace": ["  File \"/usr/lib/python3.6/unittest/case.py\", line 59, in testPartExecutor\n    yield\n", "  File \"/usr/lib/python3.6/unittest/case.py\", line 605, in run\n    testMethod()\n", "  File \"/home/du_ds/Documents/Git/PythonDataAnalysisCourse/hy-data-analysis-with-python-summer-2019/part02-e09_rational/test/test_rational.py\", line 44, in test_comparison\n    self.assertEqual(r1==r2, False, msg=\"Incorrect result for operation %s == %s!\" % (r1,r2))\n", "  File \"/home/du_ds/Documents/Git/PythonDataAnalysisCourse/hy-data-analysis-with-python-summer-2019/part02-e09_rational/src/rational.py\", line 88, in __eq__\n    a, b, c = common_denominator(self, rational2)\n"]}, {"name": "test.test_rational.RationalTest.test_plusminus", "status": "errored", "message": "name 'common_denominator' is not defined", "passed": false, "points": ["p02-09.1"], "backtrace": ["  File \"/usr/lib/python3.6/unittest/case.py\", line 59, in testPartExecutor\n    yield\n", "  File \"/usr/lib/python3.6/unittest/case.py\", line 605, in run\n    testMethod()\n", "  File \"/home/du_ds/Documents/Git/PythonDataAnalysisCourse/hy-data-analysis-with-python-summer-2019/part02-e09_rational/test/test_rational.py\", line 38, in test_plusminus\n    self.assertEqual(r1+r2, Rational(11,12), msg=\"Incorrect result for operation %s + %s!\" % (r1,r2))\n", "  File \"/home/du_ds/Documents/Git/PythonDataAnalysisCourse/hy-data-analysis-with-python-summer-2019/part02-e09_rational/src/rational.py\", line 47, in __add__\n    a1, a2, b = common_denominator(self, rational2)\n"]}, {"name": "test.test_rational.RationalTest.test_print", "status": "passed", "message": "", "passed": true, "points": ["p02-09.1"], "backtrace": []}, {"name": "test.test_rational.RationalTest.test_proddiv", "status": "errored", "message": "name 'reduce_fraction_fast' is not defined", "passed": false, "points": ["p02-09.1"], "backtrace": ["  File \"/usr/lib/python3.6/unittest/case.py\", line 59, in testPartExecutor\n    yield\n", "  File \"/usr/lib/python3.6/unittest/case.py\", line 605, in run\n    testMethod()\n", "  File \"/home/du_ds/Documents/Git/PythonDataAnalysisCourse/hy-data-analysis-with-python-summer-2019/part02-e09_rational/test/test_rational.py\", line 32, in test_proddiv\n    self.assertEqual(r1*r2, Rational(2,12), msg=\"Incorrect result for operation %s * %s!\" % (r1,r2))\n", "  File \"/home/du_ds/Documents/Git/PythonDataAnalysisCourse/hy-data-analysis-with-python-summer-2019/part02-e09_rational/src/rational.py\", line 65, in __mul__\n    a, b = reduce_fraction_fast(a, b)\n"]}]